### DISCUSSION TITLE
Guidance on adding new soil type

### DISCUSSION THREAD

**Discussion by NicWayand** [CH-Earth/summa#551] [General]
Trying to add a new soil type(s) to represent roads (asphalt, cement, etc). I am looking for guidance on the best way to do this within the available methods SUMMA gets soil parameters. Below are my suggestions:

1) Add new rows to a table in SOILPARAM.TBL. Are these tables meant to be fixed to be backward compatible with Noah-MP code? When adding a new row, what other downstream changes are required?
2) Use existing soilTypeIndex, but overwrite parameters in Param Trial.
3) ?

**Comment by andywood** [comment]
@Nic, you can alter SOILPARAM.TBL to be your own version ... add a table
name, change table params, etc.  It may be nice to have one or more
versions that are consistent with the Noah-MP table, but there's nothing
completely sacrosanct about the specific tables included there.  If you
come up with a useful alternative table, perhaps in the end it could be
merged with one of the default TBL options for SUMMA.

The TBL params do not overwrite params in Param Trial -- rather the reverse
-- so you can use the trialParams to supercede TBL values without adjusting
the original TBLs.  There's a lot of flexibility.

**Comment by NicWayand** [comment]
Thanks Andy. I took the modify SOILPARAM.TBL approach and added a new soil type that copied the Rosetta "clay" but changed theta_sat=0.030 and theta_res=0.001 (based on lit for asphalt). This leads to soil hydrology mass balance errors though. Is there any reason why I can't make the soil layers have very small porosity? 


```
solution method = 2
data_step = 3600.0000000000
totalSoilCompress = 0.7248674358
scalarTotalSoilLiq = 119.4774187422
scalarTotalSoilIce = 0.0000000000
balanceSoilWater0 = 115.3849740544
balanceSoilWater1 = 119.4774187422
balanceSoilInflux = 4.5730866114
balanceSoilBaseflow = 0.0000000000
balanceSoilDrainage = 0.0000000000
balanceSoilET = 0.0125707987
scalarSoilWatBalError = 0.2316547134
scalarSoilWatBalError = 0.0002316547
absConvTol_liquid = 0.0000100000



FATAL ERROR: run_oneGRU (gru index = 1)/run_oneHRU (hruId = 20705)/coupled_em/soil hydrology does not balance
```

**Comment by nmizukami** [comment]
Hi @NicWayand,  If I get this kind of error, I first try reducing `minstep` in localParamInfo.txt so the model can run at a shorter time step.  Changing this mostly resolve the error for me.

**Comment by andywood** [comment]
Naoki's suggestion is good, and also increasing max-iter (in local params)
can help with convergence errors.  That comes as a slight cost as you slow
the runs, however.
Also, 'asphalt' is an interesting edge case :).  In that case, would there
actually be a soil water balance?  Or should it be pre-empted by
other parameters (connoting infiltration of zero)?
Btw, you won't be able to push your modified version of SOILPARAM.TBL back
to the central repo unless it has a notable addition of possibly broader
interest.
